packages/components/src/components/Pagination/usePagination.ts

```diff
@@ -0,0 +1,75 @@
+import { useMemo } from "react";
+
+const DOTS = "...";
+// SIDE_COUNT is used to calculate how many page neighbors there are outside the current page
+const SIDE_COUNT = 2;
+// These are extra blocks to make the total blocks add up to seven
+const ADDITIONAL_BLOCK_COUNT = 2;
+// This helps to determine the total blocks
+const ACCOUNT_FOR_CURRENT_PAGE = 1;
+// These are the blocks of the first and last page that are always displayed
+const FIXED_BLOCK_COUNT = 2;
+
+// Calculates the numbers to display between two numbers
+const range = (from: number, to: number, step = 1) => {
+  let i = from;
+  const range = [];
+
+  while (i <= to) {
+    range.push(i);
+    i += step;
+  }
+
+  return range;
+};
```

Comment:
Maybe don't reinvent the wheel and use the function from loadash as we have this as dependency either way?
https://lodash.com/docs/4.17.15#range
```suggestion
```